Orange/data/table/base.py

```diff
@@ -0,0 +1,1333 @@
+    def Y(self):
+        """Return a read-only numpy array of Y.
+
+        If there is only one column, a one-dimensional array is returned. Otherwise 2D.
+        The columns are in the same order as the columns in Table.domain.class_vars.
+
+        Returns
+        -------
+        np.ndarray
+        """
+        result = self._to_numpy(Y=True)
+        return result[:, 0] if result.shape[1] == 1 else result
+
+    @property
+    def metas(self):
+        """Return a read-only 2D numpy array of metas.
+
+        The columns are in the same order as the columns in Table.domain.metas.
+
+        Returns
+        -------
+        np.ndarray
+        """
+        return self._to_numpy(meta=True)
+
+    @property
+    def weights(self):
+        """Get the weights as a 1D numpy array."""
+        if isinstance(self, SeriesBase):
+            # at least 1D when using this in TableSeries, which instead returns a 0D ndarray directly
+            result = np.atleast_1d(self[self._WEIGHTS_COLUMN])
+        else:
+            result = self[self._WEIGHTS_COLUMN].values
+        # even if weights are set with an integer, we need to return a float
+        return result.astype(float)
+
+    def set_weights(self, weight):
+        """Set the weights for the instances in this table.
+
+        Parameters
+        ----------
+        weight : Number or str or Sequence or np.ndarray or pd.Series
+            If a number, all weights are set to that value.
+            If a string, weights are set to whatever the column with that name's values are,
+            but only if those values are all numbers and are not NA/NaN.
+            If a sequence of (non-NA/NaN) numbers, set those values as the sequence.
+
+        Raises
+        ------
+        ValueError
+            If weights are nan, the column does not exist or the sequence length is mismatched.
+        TypeError
+            If an unrecognized type is passed.
+        """
+        if isinstance(weight, Number):
+            weights = weight
+        elif isinstance(weight, str):
+            if weight not in self.columns:
+                raise ValueError("{!r} is not a column.".format(weight))
+            weights = self[weight].values
+        elif isinstance(weight, (np.ndarray, Sequence)):
+            weights = np.asanyarray(weight)
+        elif isinstance(weight, pd.Series):
+            weights = weight.values
+        elif isinstance(weight, pd.Categorical):
+            weights = np.asanyarray(weight)
+        else:
+            raise TypeError("Expected one of [Number, str, Sequence, SeriesBase].")
+
+        if not isinstance(weights, Number):
+            weights = np.ravel(weights)
+            if len(weights) != len(self):
+                raise ValueError("The sequence has length {}, expected length {}.".format(len(weights), len(self)))
+
+        if bn.anynan(weights):
+            raise ValueError("Weights cannot be nan.")
+        self[self._WEIGHTS_COLUMN] = weights
```

Comment:
You are missing an important float cast before anynan check above:
```py
try:
    weights = np.asanyarray(new_weights, dtype=float)
except ValueError:
    raise ValueError("Weight values must be numbers or convertible to numbers") from None
```
Weights could have been anything. Instead, they have to be convertible to floats.